platform/platform-filter-delegate/src/main/java/org/codice/ddf/platform/filter/delegate/DelegateServletFilter.java

```diff
@@ -112,6 +97,155 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
+        }
+    }
+
+    public void addSecurityFilter(
+            final ServiceReference<SecurityFilter> securityFilterServiceReference)
+            throws ServletException {
+        if (notYetInitializedSecurityFilterServiceReferences.contains(securityFilterServiceReference)
+                || sortedAndInitializedSecurityFilterServiceReferences.contains(
+                securityFilterServiceReference)) {
+            LOGGER.warn("SecurityFilter has already been added. Not adding again.");
+        } else {
+            // only initialize SecurityFilters after the DelegateServletFilter has been initialized with a filterConfig
+            if (hasBeenInitialized) {
+                initAndAddToFilterList(securityFilterServiceReference, filterConfig);
+            } else {
+                notYetInitializedSecurityFilterServiceReferences.add(securityFilterServiceReference);
+            }
+        }
+    }
+
+    private void initAndAddToFilterList(
+            ServiceReference<SecurityFilter> securityFilterServiceReference,
+            FilterConfig filterConfig) throws ServletException {
+        final ServletContext servletContext;
+
+        if (filterConfig == null) {
+            LOGGER.error(
+                    "DelegateServletFilter was initialized with a null filterConfig. Initializing SecurityFilter {} with null ServletContext",
+                    ServicePropertiesUtils.getStringProperty(securityFilterServiceReference,
+                            HttpWhiteboardConstants.HTTP_WHITEBOARD_FILTER_NAME));
+            servletContext = null;
+        } else {
+            servletContext = this.filterConfig.getServletContext();
+        }
+
+        bundleContext.getService(securityFilterServiceReference)
+                .init(new SecurityFilterInitFilterConfig(securityFilterServiceReference,
+                        servletContext));
+        sortedAndInitializedSecurityFilterServiceReferences.add(securityFilterServiceReference);
+
+        LOGGER.debug("Initialized and added SecurityFilter {}",
+                ServicePropertiesUtils.getStringProperty(securityFilterServiceReference,
+                        HttpWhiteboardConstants.HTTP_WHITEBOARD_FILTER_NAME));
+
+        LOGGER.debug("The SecurityFilters list is now");
+        for (ServiceReference<SecurityFilter> filter : sortedAndInitializedSecurityFilterServiceReferences) {
+            LOGGER.debug("\t{} {}",
+                    filter.getProperty(Constants.SERVICE_RANKING),
+                    ServicePropertiesUtils.getStringProperty(filter,
+                            HttpWhiteboardConstants.HTTP_WHITEBOARD_FILTER_NAME));
+        }
+    }
+
+    public void removeSecurityFilter(
+            final ServiceReference<SecurityFilter> securityFilterServiceReference) {
+        sortedAndInitializedSecurityFilterServiceReferences.remove(securityFilterServiceReference);
+        bundleContext.getService(securityFilterServiceReference)
```

Comment:
I added ` destroy-method="destroy"` to every `SecurityFilter` blueprint definition instead of calling `destroy` inside of `removeSecurityFilter`.